### ISSUE TITLE
Is it possible to generate report at the end of each Pytest session

### ISSUE THREAD

**Issue by barnwal-paras** [pytest-dev/pytest-html#256]
In my Automation Suite, I am running multiple session, but single HTML report is getting generated which contains test report of all the session combined. Is there a way to generate report for each session individually ?

**Comment by BeyondEvil** [comment]
@barnwal-paras When you say "multiple session", do you mean using xdist or multiple calls to pytest or something else?

**Comment by barnwal-paras** [comment]
@BeyondEvil   No, I am using a parametrised session scoped fixture and wanted to generate report at the end of every session, as a part of teardown of that fixture.

**Comment by BeyondEvil** [comment]
@barnwal-paras Not sure I follow. Could you give an example in code?

Do you want a report for each session?

**Comment by barnwal-paras** [comment]
```
@pytest.fixture(scope="session", params=[session1_config, session2_config])
def config(request, chrome_config):
    setup()
    yield 
    teardown()
```
I want report to be generated everytime teardown is called for those set of test having same parameter.

**Comment by BeyondEvil** [comment]
Similar to #254, this would require a major refactor due to the fact that today, the entire report is done in memory until the `pytest_sessionfinish` hook is called - that's when the writing of the report to disk happens.

One option would be for you to import the [`HTMLReport`](https://github.com/pytest-dev/pytest-html/blob/master/pytest_html/plugin.py#L91) class yourself, and add the logic necessary to generate a report for each session.
